### PR TITLE
Plott PCA of combined TOB-WGS + SNP-chip data 

### PR DESCRIPTION
**Pull request by KatalinaBobowik**

### CODE REVIEW

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        .when(
+            (
+                scores.snp_chip  # noqa: E501; pylint: disable=singleton-comparison;
+                == False  # noqa: E712
+            ),
+            'tob_wgs',
+        )
+        .default('NA')
+    )
+    scores = scores.annotate(cohort_sample_codes=expr)
```
Comment (lgruen) [inline scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py:40]: Do you need the boolean annotation first?

```suggestion
    scores = scores.annotate(cohort_sample_codes=hl.if_else(scores.s.contains('snp_chip'), 'snp_chip', 'tob_wgs'))
```

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            title='TOB-WGS + TOB SNP Chip',
+            xlabel='PC' + str(pc1 + 1) + ' (' + str(variance[pc1]) + '%)',
+            ylabel='PC' + str(pc2 + 1) + ' (' + str(variance[pc2]) + '%)',
+            hover_fields=hover_fields,
+        )
+        plot_filename = output_path('pc' + str(pc2) + '.png', 'web')
+        with hl.hadoop_open(plot_filename, 'wb') as f:
+            get_screenshot_as_png(p).save(f, format='PNG')
+        plot_filename_html = 'pc' + str(pc2) + '.html'
+        output_file(plot_filename_html)
```
Comment (lgruen) [inline scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py:72]: Did `file_html` not work?
(https://centrepopgen.slack.com/archives/D01MZL1FGDB/p1623990732019100)

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
+"""Create PCA plots for the combined
+TOB-WGS/SNP-chip data"""
```
Comment (lgruen) [inline scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py:2]:
```suggestion
"""Create PCA plots for the combined TOB-WGS/SNP-chip data"""
```

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import subprocess
+from bokeh.io.export import get_screenshot_as_png
+from bokeh.plotting import output_file, save
+import pandas as pd
+import hail as hl
+import click
+from analysis_runner import output_path
+
+SCORES = 'gs://cpg-tob-wgs-main/tob_wgs_snp_chip_variant_pca/v6/scores.ht/'
+EIGENVALUES = 'gs://cpg-tob-wgs-main/tob_wgs_snp_chip_variant_pca/v6/eigenvalues.ht'
```
Comment (lgruen) [inline scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py:13]: Do you want to use `bucket_path` here?

```suggestion
SCORES = 'gs://cpg-tob-wgs-main/tob_wgs_snp_chip_variant_pca/v6/scores.ht/'
EIGENVALUES = 'gs://cpg-tob-wgs-main/tob_wgs_snp_chip_variant_pca/v6/eigenvalues.ht'
```

**Review by lgruen — CHANGES_REQUESTED** [review]

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py (hunk 2 above)
Comment (KatalinaBobowik) [reply scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py:54]: the above method was the only way I could get it to work for me

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py (hunk 4 above)
Comment (KatalinaBobowik) [reply scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py]: I changed this to use the `bucket_path()` convenience function

File: scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py (hunk 1 above)
Comment (KatalinaBobowik) [reply scripts/plotting/tob_wgs_snp_chip_plot_pca/plot_pca_tob_wgs_snp_chip_datasets.py]: This is a much better way than what I was doing before!

**Review by lgruen — APPROVED** [review]

**Pull request merged**